src/main/webapp/profileScript.js

```diff
@@ -1,37 +1,49 @@
 /*
- * Loads certain buttons in the navbar based on the user's login status.
+ * Updates the page's display based on the user's login status.
  */
-function getLoginStatus() {
+function loginStatusDisplay() {
+  // Clear the user-specific elements previously displayed.
   const userNavbarSection = document.getElementById('user-navbar-section');
   userNavbarSection.innerHTML = '';
 
+  // Get the login status and display the corresponding elements.
   const promise = fetch('/login').then(response => response.json()).then((json) => {
-
-    // If the user is logged in, add a logout and profile button.
     if (json['loginStatus'].localeCompare('true') == 0) {
-      const logoutButton = document.createElement('button');
-      logoutButton.innerText = 'Logout';
-      logoutButton.addEventListener('click', () => {
-        window.location.href = json['logoutUrl'];
-      });
-
-      const personalProfileButton = document.createElement('button');
-      personalProfileButton.innerText = 'My Profile';
-      personalProfileButton.addEventListener('click', () => {
-        window.location.href = 'profile.html';
-      });
-
-      userNavbarSection.appendChild(logoutButton);
-      userNavbarSection.appendChild(personalProfileButton);
-
-    // If the user is logged out, add a login button.
+      addLoggedInDisplay();
     } else {
-      const loginButton = document.createElement('button');
-      loginButton.innerText = 'Login';
-      loginButton.addEventListener('click', () => {
-        window.location.href = json['loginUrl'];
-      });
-      userNavbarSection.appendChild(loginButton);
+      addLoggedOutDisplay();
     }
   });
 }
+
+/*
+ * Adds the elements that should be displayed when the user is logged in.
+ */
+function addLoggedInDisplay() {
```

Comment:
Undid these changes of separating into functions because realized actions were too intertwined with original function, more efficient just to leave it as one.